josemoren/pycharm-pypendency-plugin
# Add action to go to code from DI file

This PR adds a new action to go from the DI file to its corresponding source code, and renames the former `Pypendency` action to `Pypendency: Go to/Create DI file`. For ease of use, **I personally bind both actions to the same keymap**

<img width="494" alt="image" src="https://github.com/josemoren/pycharm-pypendency-plugin/assets/53608787/5b53ba2c-c299-4350-816b-b08e9a687a0f">

### How should this be tested? 🤔 
- Go to a `.py` DI file, run the new action `Pypendency: Go to code` and check it opens the source code
- Go to a `.yaml/yml` DI file and check that the behaviour is the same

### Note 🗒️ 
The old action, now renamed to `Pypendency: Go to/Create DI file`, behaves differently depending on whether it's being called from a DI file or from a source code file ([see](https://github.com/josemoren/pycharm-pypendency-plugin/blob/81ed14b13e5ae557921cc7e7133bc0b5ba17d3ff/src/main/java/org/fever/GotoPypendencyOrCodeAction.java#L25-L29)). This is a workaround to **allowing devs to set the same keymap for both actions**, for going from-to DI files with a single shortcut. If we weren't to add this, PyCharm would try to create a new DI file whenever we call the shortcut from a .py DI file
